projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/vdk_jupyterlab_extension/vdk_ui.py

```diff
@@ -156,3 +157,57 @@ def create_deployment(
             enabled=enabled,
         )
         return output
+
+    @staticmethod
+    def get_failing_notebook_info(failing_cell_id: str, job_path: str):
+        """
+        Return information about the notebook where a given cell is causing a job to fail.
+        :param failing_cell_id: the id of the cell that is causing the 'Job Run' to fail
+        :param job_path: the path to the job's directory
+        :return: path of the notebook where the failing cell is located, index of the failing cell
+             if the specified job path does not exist, an empty dictionary is returned.
+        """
+        if not os.path.exists(job_path):
+            job_path = os.getcwd() + job_path
+            if not os.path.exists(job_path):
+                return ""
+        notebook_files = [
+            x
+            for x in pathlib.Path(job_path).iterdir()
+            if (x.name.lower().endswith(".ipynb"))
+        ]
+        for notebook_file in notebook_files:
+            with open(notebook_file) as f:
+                notebook_json = json.load(f)
+                cell_index = 0
+                for jupyter_cell in notebook_json["cells"]:
+                    if jupyter_cell["id"] == failing_cell_id:
```

Comment:
nit: why looping over the entire list of cells, while having the exact cell_id at hand?

Reply:
It is not possible to get the cell with the specified ID without looping through the cells in the notebook JSON object.
This is because the notebook JSON object stores cells as a list, and there is no direct way to access a specific cell in the list without iterating over the cells. The cell ID is not an index in the list, but rather an identifier that is stored within each cell's metadata.

Reply:
I don't understand why looping over aall the notebooks in potential existence is needed, to understand what notebook the failing cell comes from?

May you provide an example of two short notebook JSON cells?

Reply:
I see, the notebook is a composite of cells. However, having a cell_id str, why do we still need to loop over aall the notebooks and cells to find where it belongs to? 
What if the cell_id contains the information about the notebook it belongs to, so we don't have to parse alll the notebooks' cells?

Reply:
I do not think we can get all that information from the front-end, since Jupyter uses another path system (it does not show the full path but only part of it) it becomes pretty complex and I think this way is a lot simple